langchain_lakefs/document_loaders.py

```diff
@@ -1,72 +1,161 @@
-    # https://github.com/langchain-ai/langchain/blob/869523ad728e6b76d77f170cce13925b4ebc3c1e/libs/community/langchain_community/document_loaders/recursive_url_loader.py#L54
     """
-    LakeFS document loader integration
+    LakeFS document loader integration using the official lakeFS Python SDK.
 
-    # TODO: Replace with relevant packages, env vars.
     Setup:
-        Install ``langchain-lakefs`` and set environment variable ``LAKEFS_API_KEY``.
+        Install ``langchain-lakefs`` and ``lakefs``:
 
         .. code-block:: bash
 
-            pip install -U langchain-lakefs
-            export LAKEFS_API_KEY="your-api-key"
+            pip install -U langchain-lakefs lakefs-client
 
-    # TODO: Replace with relevant init params.
     Instantiate:
         .. code-block:: python
 
             from langchain_community.document_loaders import LakeFSLoader
 
             loader = LakeFSLoader(
-                # required params = ...
-                # other params = ...
+                lakefs_endpoint="https://example.my-lakefs.com",
+                lakefs_access_key="your-access-key",
+                lakefs_secret_key="your-secret-key",
+                repo="my-repo",
+                ref="main",
+                path="path/to/files"
             )
+    """
 
-    Lazy load:
-        .. code-block:: python
-
-            docs = []
-            docs_lazy = loader.lazy_load()
-
-            # async variant:
-            # docs_lazy = await loader.alazy_load()
-
-            for doc in docs_lazy:
-                docs.append(doc)
-            print(docs[0].page_content[:100])
-            print(docs[0].metadata)
-
-        .. code-block:: python
-
-            TODO: Example output
-
-    # TODO: Delete if async load is not implemented
-    Async load:
-        .. code-block:: python
-
-            docs = await loader.aload()
-            print(docs[0].page_content[:100])
-            print(docs[0].metadata)
-
-        .. code-block:: python
-
-            TODO: Example output
-    """  # noqa: E501
-
-    # TODO: This method must be implemented to load documents.
-    # Do not implement load(), a default implementation is already available.
-    def lazy_load(self) -> Iterator[Document]:
-        raise NotImplementedError()
-
-    # TODO: Implement if you would like to change default BaseLoader implementation
-    # async def alazy_load(self) -> AsyncIterator[Document]:
+    def __init__(
+            self,
+            lakefs_endpoint: str,
+            lakefs_access_key: str,
+            lakefs_secret_key: str,
+            repo: str,
+            ref: str = "main",
+            path: str = "",
+    ):
+        self.client = Client(
+            host=lakefs_endpoint,
+            username=lakefs_access_key,
+            password=lakefs_secret_key
+        )
+        self.repo = repo
+        self.ref = ref
+        self.path = path
+
+    def set_path(self, path: str) -> None:
+        self.path = path
```

Comment:
Add code document when possible

```suggestion
    def set_path(self, path: str) -> None:
        """Set the path to load documents from."""
        self.path = path
```